lib/puppet/provider/exec.rb

```diff
@@ -4,71 +4,71 @@
 class Puppet::Provider::Exec < Puppet::Provider
```

Comment:
This is just cherry-picking 9a829ae4a98152d67246cca4e6bd44ebae94dbde -- I changed the commit message as the context in the original message no longer applies.

Reply:
TODO: Update the commit message + mention the cherry-picking